### PR TITLE
Consider trivia when testing code fixes and refactorings

### PR DESCRIPTION
**Pull request by sharwell**
* Change `ignoreTrivia` to default to false
* Fix tests so the expected test results accurately reflect reality, including trivia

**Reviewers:** The main item to verify is that this pull request only changes expected test outputs. Test inputs should remain the same. One (seemingly unavoidable) exception is indicated with a ⚠️ comment.

:memo: The changes to expected test results reveal a large number of bugs. These will be identified in comments in this PR, and bugs filed so they can be fixed later.

### CODE REVIEW

File: src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateEndConstruct/GenerateEndConstructTests.vb [hunk 1]
```diff
@@ -45,7 +45,7 @@ End Using</MethodBody>
             Dim text = <File>
 Structure Goo[||]</File>
 
-            Dim expected = StringFromLines("Structure Goo", "End Structure", "")
+            Dim expected = StringFromLines("", "Structure Goo", "End Structure", "")
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateEndConstruct/GenerateEndConstructTests.vb]: 😕 Why does this one not produce a blank line in the output line the others below do? I'll file a bug, but not sure if it's something to change.

File: src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateMethod/GenerateMethodTests.vb [hunk 2]
```diff
@@ -297,13 +323,14 @@ End Class")
...
 End Class",
 "Imports System
+
 Class C
     Sub M(i As Integer)
-        Goo(NextMethod)
-    End Sub
-    Private Sub Goo(nextMethod As Object)
+        Goo(NextMethod) End Sub 
+Private Sub Goo(nextMethod As Object)
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateMethod/GenerateMethodTests.vb]: :bug: Incorrect indentation, and should have been preceded by a blank line.

File: src/EditorFeatures/VisualBasicTest/Diagnostics/InsertMissingCast/InsertMissingCastTests.vb [hunk 3]
```diff
@@ -113,8 +113,8 @@ Module M1
     Sub goo(d As Integer)
     End Sub
     Sub Main()
-        goo(CInt(""10"")) 
- End Sub
+        goo(CInt(""10""))
+    End Sub
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/InsertMissingCast/InsertMissingCastTests.vb]: :bug: Should not have indented `End Sub` here.

File: src/EditorFeatures/VisualBasicTest/Diagnostics/InsertMissingCast/InsertMissingCastTests.vb [hunk 4]
```diff
@@ -149,8 +149,8 @@ End Module",
 "Option Strict On
 Module M1
     Function goo() As Integer
-        Return CInt(""10"") 
- End Function
+        Return CInt(""10"")
+    End Function
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/InsertMissingCast/InsertMissingCastTests.vb]: :bug: Should not have indented `End Function` here (another case of previous bug).

File: src/EditorFeatures/VisualBasicTest/Diagnostics/PreferFrameworkType/PreferFrameworkTypeTests.vb [hunk 5]
```diff
@@ -381,10 +392,11 @@ End Class
 ",
 "Imports System
 Class C
-    ''' <see cref=""Int32.MaxValue""/>
+    ''' <see cref=""Integer.MaxValue""/>
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/PreferFrameworkType/PreferFrameworkTypeTests.vb]: :bug:

File: src/EditorFeatures/VisualBasicTest/Diagnostics/UseAutoProperty/UseAutoPropertyTests.vb [hunk 6]
```diff
@@ -86,7 +86,8 @@ end class")
     end property|]
 end class",
 "class Class1
-    readonly property P as integer = 1
+    readonly property P as integer
+= 1
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/UseAutoProperty/UseAutoPropertyTests.vb]: :bug: Formatting

File: src/EditorFeatures/VisualBasicTest/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.vb [hunk 7]
```diff
@@ -90,11 +90,14 @@ index:=1, ignoreTrivia:=False)
...
 "Imports System.Collections.Generic
+
 Partial Class c1(Of V As {New}, U)
     Dim x As New V
+
     Public Overrides Function Equals(obj As Object) As Boolean
         Dim c = TryCast(obj, c1(Of V, U))
-        Return c IsNot Nothing AndAlso EqualityComparer(Of V).Default.Equals(x, c.x)
+        Return c IsNot Nothing AndAlso
+               EqualityComparer(Of V).Default.Equals(x, c.x)
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.vb]: :bug: This formatting is not consistent with the smart indent implementation. The second line should be indented 4 spaces, not 7.

File: src/EditorFeatures/VisualBasicTest/MoveDeclarationNearReference/MoveDeclarationNearReferenceTests.vb [hunk 8]
```diff
@@ -255,7 +256,7 @@ using System.Linq
 class Program
     sub M()
         for each (v in x)
-            {|Warning:dim i = CInt(0)|}
+        {|Warning:dim i = CInt(0)|}
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/MoveDeclarationNearReference/MoveDeclarationNearReferenceTests.vb]: :bug: Incorrect indentation

File: src/EditorFeatures/VisualBasicTest/MoveDeclarationNearReference/MoveDeclarationNearReferenceTests.vb [hunk 9]
```diff
@@ -123,7 +123,8 @@ end class",
 "class C
     sub M()
         if true
-            dim x as integer = 5
+            dim x as integer
+= 5
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/MoveDeclarationNearReference/MoveDeclarationNearReferenceTests.vb]: :bug: Should not have split the line.

File: src/EditorFeatures/VisualBasicTest/RemoveUnnecessaryImports/RemoveUnnecessaryImportsTests.vb [hunk 10]
```diff
@@ -417,8 +417,7 @@ Namespace goo
     Class bar
     End Class
 End Namespace|]",
-"Imports System, System.Collections.Generic
-Module Program
+"Imports System, System.Collections.Generic Module Program
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/RemoveUnnecessaryImports/RemoveUnnecessaryImportsTests.vb]: :bug: `Module` should still appear on the next line.

File: src/EditorFeatures/VisualBasicTest/RemoveUnnecessaryImports/RemoveUnnecessaryImportsTests.vb [hunk 11]
```diff
@@ -516,7 +515,7 @@ Namespace B
     Public Class CB
     End Class
 End Namespace|]",
-"Imports A
+"Imports A _
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/RemoveUnnecessaryImports/RemoveUnnecessaryImportsTests.vb]: :bug: Should omit `_`.

Comment (alrz) [inline src/EditorFeatures/VisualBasicTest/RemoveUnnecessaryImports/RemoveUnnecessaryImportsTests.vb]: Curious, do you use some tool to update assertions based on the actual output?

Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/RemoveUnnecessaryImports/RemoveUnnecessaryImportsTests.vb]: No, I manually reviewed each case. 😦

File: src/EditorFeatures/VisualBasicTest/Utils.vb [hunk 12]
```diff
@@ -327,7 +327,7 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests
                         "End Structure"
 
                 Case "File"
-                    Return String.Join(vbCrLf, lines)
+                    Return testSource.NormalizedValue
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/Utils.vb]: ⚠️ This causes some test inputs to change (addition of blank lines at the beginning and/or end of the input), but I didn't see a clean way to separate the two.

File: src/EditorFeatures/VisualBasicTest/RemoveUnnecessaryImports/RemoveUnnecessaryImportsTests.vb (hunk 11 above)
Comment (alrz) [inline src/EditorFeatures/VisualBasicTest/RemoveUnnecessaryImports/RemoveUnnecessaryImportsTests.vb]: Currently my bottleneck is the test-runner - tried a couple of other tools. It's really inconvenient to use command-line or wpf xunit runner. I'm wondering if either of those are what you're using for local testing.

Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/RemoveUnnecessaryImports/RemoveUnnecessaryImportsTests.vb]: I used the WPF runner.

Comment (alrz) [inline src/EditorFeatures/VisualBasicTest/RemoveUnnecessaryImports/RemoveUnnecessaryImportsTests.vb]: That's great! I'm looking forward to it. Thanks.

**Comment by CyrusNajmabadi** [comment]
>Change ignoreTrivia to default to false

Any need to have the option at all now?  I think it ahs served its purpose.  It was originally used because of too much churn in the formatter, but now that that has stabilized, i would be fine with it going away.

**Comment by sharwell** [comment]
> Any need to have the option at all now? I think it ahs served its purpose. It was originally used because of too much churn in the formatter, but now that that has stabilized, i would be fine with it going away.

@CyrusNajmabadi  The PR is unmanageable with that change merged with this one.

**Comment by alrz** [comment]
> Then we can look at removing it.

It is very useful when you're just verifying the behavior without worrying about trivia. After that it should be required to be false (or absent with default=false) before merge.

File: src/EditorFeatures/Test/CodeGeneration/CodeGenerationTests.VisualBasic.cs [hunk 13]
```diff
@@ -710,31 +835,48 @@ End Class
...
                 await TestAddStatementsAsync(input, expected, "Console.WriteLine(\"Setting the value\"");
             }
 
             [Fact, Trait(Traits.Feature, Traits.Features.CodeGeneration)]
             public async Task AddParametersToMethod()
             {
                 var input = "Class C\n Public [|Sub M()\n End Sub|]\n End Class";
-                var expected = "Class C\n Public Sub M(num As Integer, Optional text As String = \"Hello!\", Optional floating As Single = 0.5)\n End Sub\n End Class";
+                var expected = @"Class C
+ Public Sub M(numAs Integer, OptionaltextAs String = ""Hello!"",OptionalfloatingAs Single = 0.5)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test/CodeGeneration/CodeGenerationTests.VisualBasic.cs]: I don't think the codegenerator does any formatting.  You'd need to pass hte node to a formatter to do that sort of thing.

**Comment by CyrusNajmabadi** [comment]
> It is very useful when you're just verifying the behavior without worrying about trivia. After that it should be required to be false

Seems reasonable to me.

File: src/EditorFeatures/CSharpTest/CodeActions/ExtractMethod/ExtractMethodTests.cs [hunk 14]
```diff
@@ -151,8 +151,8 @@ static void Main(string[] args)
 
     private static bool NewMethod(bool b)
     {
-        return b != 
-            true;
+        return b !=
+                    true;
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/CodeActions/ExtractMethod/ExtractMethodTests.cs]: :arrow_right: #21453

File: src/EditorFeatures/CSharpTest/CodeActions/InlineTemporary/InlineTemporaryTests.cs [hunk 15]
```diff
@@ -15,7 +15,7 @@ public class InlineTemporaryTests : AbstractCSharpCodeActionTest
         protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
             => new InlineTemporaryCodeRefactoringProvider();
 
-        private async Task TestFixOneAsync(string initial, string expected, bool ignoreTrivia = true)
+        private async Task TestFixOneAsync(string initial, string expected, bool ignoreTrivia = false)
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/CodeActions/InlineTemporary/InlineTemporaryTests.cs]: ➡️ #21454

File: src/EditorFeatures/CSharpTest/CodeActions/InlineTemporary/InlineTemporaryTests.cs [hunk 16]
```diff
@@ -1366,15 +1377,14 @@ static void Main(string[] args)
 
     int a = x;
 }",
-                       @"{
+@"
+{
         int y,
-#if true
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/CodeActions/InlineTemporary/InlineTemporaryTests.cs]: :arrow_right: #21455

File: src/EditorFeatures/CSharpTest/CodeActions/InlineTemporary/InlineTemporaryTests.cs [hunk 17]
```diff
@@ -1391,15 +1401,14 @@ static void Main(string[] args)
 
     int a = x;
 }",
-                       @"{
+@"
+{
         int y,
 #if true
-            z
-#endif
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/CodeActions/InlineTemporary/InlineTemporaryTests.cs]: :arrow_right: https://github.com/dotnet/roslyn/issues/21455#issuecomment-321914400

File: src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs [hunk 18]
```diff
@@ -139,12 +144,13 @@ class Class2 { }";
             var codeAfterMove =
 @"// Banner Text
 using System;
-
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs]: :arrow_right: #21456

File: src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs [hunk 19]
```diff
@@ -772,10 +779,12 @@ class Outer {
     }
 }";
             var codeAfterMove = @"
+// Only used by inner type.
+
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs]: :arrow_right: #21459

File: src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs [hunk 20]
```diff
@@ -204,7 +210,6 @@ int Goo
 }",
 @"class C
 {
-#if true
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs]: :arrow_right: #21460

File: src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs [hunk 21]
```diff
@@ -737,6 +742,7 @@ void SetGoo(int i)
 
 class C
 {
+
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs]: :arrow_right: #21461.

**Pull request merged**